Documentation/troubleshooting.rst

```diff
@@ -137,15 +142,53 @@ e.g.:
 Connectivity Problems
 =====================
 
+Cilium connectivity tests
+------------------------------------
+
+The Cilium connectivity test_ deploys a series of services, deployments, and 
```

Comment:
Two comments:

* Did you intend to explicitly describe to the user how to deploy these? That's at least how the other docs reference the connectivity-check YAML. Note that there is already a template of that text in `Documentation/gettingstarted/k8s-install-connectivity-test.rst`. We should be able to just import that docs fragment here.

* Secondly, this will only work in a namespace by itself with no other pods or network policies applied. If there is a Cilium Clusterwide Network Policy enabled, that may also break this connectivity check.

Reply:
Thanks, Joe. 
> * Did you intend to explicitly describe to the user how to deploy these? That's at least how the other docs reference the connectivity-check YAML. Note that there is already a template of that text in `Documentation/gettingstarted/k8s-install-connectivity-test.rst`. We should be able to just import that docs fragment here.

This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template. 

> * Secondly, this will only work in a namespace by itself with no other pods or network policies applied. If there is a Cilium Clusterwide Network Policy enabled, that may also break this connectivity check.

That's a great point. Given that the test deploys to the default namespace, this could cause problems in a production environment. We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like `cilium-test`?). How is this generally handled in the wild today? Do we add the `Namespace` stanza or avoid suggesting these tests in production environments?

Reply:
> This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template.

Are we able to share more here? I'm mainly concerned about when we inevitably make changes to these common fragments, and have to go and update multiple places. Whereas if we're able to tie it down to the one fragment file, we can just fix up the one example.

> We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like cilium-test?). How is this generally handled in the wild today? Do we add the Namespace stanza or avoid suggesting these tests in production environments?

I like the `cilium-test` namespace idea. We may consider actually just changing the namespace in the YAMLs themselves to achieve this.

As for today, I think we just have the instructions in the "getting started guides" and infer that users won't run them in a sensitive namespace.

Reply:
> > This section is a slightly modified copy of that docs fragment. I updated only to describe what gets deployed but the rest of the text is verbatim from that template.
> 
> Are we able to share more here? I'm mainly concerned about when we inevitably make changes to these common fragments, and have to go and update multiple places. Whereas if we're able to tie it down to the one fragment file, we can just fix up the one example.
> 

I like this idea. I'm going to develop a breakdown of each test and specify what type of traffic it's validating. I'll update the testing definitions here and we can decide where one fragment will live and can be referenced. Let me know if you think I should move it to a dedicated location. 

> > We could either communicate that caveat or change the namespace that it's deployed to (something unlikely to be in use like cilium-test?). How is this generally handled in the wild today? Do we add the Namespace stanza or avoid suggesting these tests in production environments?
> 
> I like the `cilium-test` namespace idea. We may consider actually just changing the namespace in the YAMLs themselves to achieve this.
> 
> As for today, I think we just have the instructions in the "getting started guides" and infer that users won't run them in a sensitive namespace.

That inference makes sense for a getting started guide, but once this is published to a troubleshooting guide, we should be more clear on the impacts the testing could have on their system. We can give a high-level caveat about cluster-wide network policies, but I'm curious about how you think we should handle the namespacing. Do you think we should adjust the namespace this is deployed to directly in the yaml or for the purposes of this doc we can have the operator create the cilium-test namespace and deploy with `k apply --namespace=cilium-test -f https://..connectivity-check.yaml`?